src/android/java/io/jxcore/node/BtToRequestSocket.java

```diff
@@ -67,75 +63,64 @@ public BtToRequestSocket(BluetoothSocket socket, BtSocketDisconnectedCallBack ha
+        try {
+            if (readyCallback != null) {
+                readyCallback.listeningAndAcceptingNow(mHTTPPort);
             }
+            print_debug("Now accepting connections");
+            Socket tmpSocket = srvSocket.accept();
+            CloseSocketAndStreams();
+            localHostSocket = tmpSocket;
 
-            if(!mStopped) {
-                print_debug("Set local streams");
-                LocalInputStream = tmpInputStream;
-                LocalOutputStream = tmpOutputStream;
-
-                if (mmInStream != null && LocalInputStream != null
-                 && mmOutStream != null && LocalOutputStream != null
-                 && localHostSocket != null) {
-
-                    SendingThread = new StreamCopyingThread(this, LocalInputStream, mmOutStream);
-                    SendingThread.setDebugTag("--Sending");
-                    SendingThread.start();
-
-                    ReceivingThread = new StreamCopyingThread(this, mmInStream, LocalOutputStream);
-                    ReceivingThread.setDebugTag("--Receiving");
-                    ReceivingThread.start();
-
-                    print_debug("Stream Threads are running");
-                } else {
-                    mStopped = true;
-                    print_debug("at least one stream is null");
-                    mHandler.Disconnected(that,"at least one stream is null");
-                }
-            }
-        } else {
+            print_debug("incoming data from: " + GetLocalHostAddressAsString() + ", port: " + GetLocalHostPort());
+            tmpInputStream = localHostSocket.getInputStream();
+            tmpOutputStream = localHostSocket.getOutputStream();
+
+        } catch (Exception e) {
             mStopped = true;
-            print_debug("creating  socket failed");
-            mHandler.Disconnected(that,"creating socket failed");
+            print_debug("Creating local streams failed: " + e.toString());
+            mHandler.Disconnected(that, "Creating local streams failed");
+            return;
         }
 
+        print_debug("Set local streams");
+        LocalInputStream = tmpInputStream;
+        LocalOutputStream = tmpOutputStream;
+
+        if (mmInStream == null || LocalInputStream == null || mmOutStream == null || LocalOutputStream == null || localHostSocket == null) {
+            mStopped = true;
+            print_debug("at least one stream is null");
+            mHandler.Disconnected(that, "at least one stream is null");
+            return;
+        }
+
+        SendingThread = new StreamCopyingThread(this, LocalInputStream, mmOutStream);
+        SendingThread.setDebugTag("--Sending");
+        SendingThread.start();
+
+        ReceivingThread = new StreamCopyingThread(this, mmInStream, LocalOutputStream);
+        ReceivingThread.setDebugTag("--Receiving");
+        ReceivingThread.start();
+
         print_debug("rin ended ---------------------------;");
     }
 
```

Comment:
In line 148 below where you have the else in StreamCopyError shouldn't it also call mHandler.Disconnected? It seems like Disconnected is your flag that something awful has happened and getting called on StreamCopyError with an unknown thread would seem to qualify as horrible.